src/backend/distributed/executor/multi_client_executor.c

```diff
@@ -815,8 +814,19 @@ MultiClientCreateWaitInfo(int maxConnections)
 {
 	WaitInfo *waitInfo = palloc(sizeof(WaitInfo));
 
+	/* we subtract 2 to make room for the WL_POSTMASTER_DEATH and WL_LATCH_SET events */
+	if (maxConnections > FD_SETSIZE - 2)
+	{
+		maxConnections = FD_SETSIZE - 2;
```

Comment:
Do we need to do this when poll is available?

Reply:
Addressed in https://github.com/citusdata/citus/pull/2011/commits/114414d9f1f75e61476d2d13493ecb3668ab1505